sdk/keyvault/keyvault-certificates/src/index.ts

```diff
@@ -869,45 +873,59 @@ export class CertificatesClient {
   }
 
   /**
-   * Creates a new certificate. If this is the first version, the certificate resource is created. This operation requires the certificates/create permission.
+   * Creates a new certificate through a Long Running Operation poller that allows you to wait indifinetly until the certificate is signed.
    *
    * Example usage:
    * ```ts
    * const client = new CertificatesClient(url, credentials);
-   * await client.createCertificate("MyCertificate", {
+   * const poller = await client.createCertificate("MyCertificate", {
    *   issuerName: "Self",
    *   subjectName: "cn=MyCert"
    * });
+   *
+   * // Serializing the poller
+   * const serialized = poller.toJSON();
+   * // A new poller can be created with:
+   * // await client.createCertificate("MyCertificate", policy, {}, { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const certificate = await poller.done();
+   * console.log(certificate);
    * ```
-   * @summary Creates a certificate
+   * @summary Creates a certificate through a Long Running Operation Poller
    * @param name The name of the certificate
    * @param certificatePolicy The certificate's policy
-   * @param [options] Optional parameters
+   * @param [createCertificateOptions] Optional parameters to the createCertificate operation
+   * @param [pollerOptions] Optional parameters to the creation of the poller
    */
   public async createCertificate(
     name: string,
     certificatePolicy: CertificatePolicy,
-    options: CreateCertificateOptions = {}
-  ): Promise<Certificate> {
-    const span = this.createSpan("createCertificate", options);
-
-    let result: CreateCertificateResponse;
+    createCertificateOptions: CreateCertificateOptions = {},
+    pollerOptions: {
+      manual?: boolean;
+      intervalInMs?: number;
+      resumeFrom?: string;
+    } = {}
+  ): Promise<CreateCertificatePoller> {
+    // Making a fake client to reduce the duplicated code
+    const pollerKeyVaultClient: CertificatesClientInterface = {
+      beginCreateCertificate: this.beginCreateCertificate.bind(this),
+      getCertificateOperation: this.getCertificateOperation.bind(this),
+      cancelCertificateOperation: this.cancelCertificateOperation.bind(this),
+      getCertificateWithPolicy: this.getCertificateWithPolicy.bind(this)
+    };
 
-    try {
-      result = await this.client.createCertificate(this.vaultBaseUrl, name, {
-        ...this.setParentSpan(span, options.requestOptions || {}),
-        certificateAttributes: {
-          ...options.certificateAttributes,
-          enabled: options.enabled
-        },
-        tags: options.tags,
-        certificatePolicy: toCorePolicy(certificatePolicy)
-      });
-    } finally {
-      span.end();
-    }
+    const poller = new CreateCertificatePoller({
+      client: pollerKeyVaultClient,
+      name,
+      certificatePolicy,
+      createCertificateOptions,
+      ...pollerOptions
+    });
```

Comment:
The poller should be created after we have already made the request to create the certificate.

Reply:
This happens inside the poller's operation! 🌞